### DISCUSSION TITLE
gatsbyjs.org's very low Lighthouse score

### DISCUSSION THREAD

**Discussion by Undistraction** [gatsbyjs/gatsby#29097] [Help]
I was chatting another developer yesterday who was considering what tech to use for a new site. We got to talking about Gatsby (vs NextJs) and I encouraged him to use Gatsby. A little while after we chatted he sent me this (from [Google Page Speed Insights](https://developers.google.com/speed/pagespeed/insights/)):

<img width="789" alt="Screenshot 2021-01-20 at 09 11 17" src="https://user-images.githubusercontent.com/161489/105153208-14154d00-5b00-11eb-9669-7662c794137c.png">

He asked me why Gatsby's own site performed so poorly. I don't really have a good answer, so I thought I'd raise it here rather than adding it to what has become an [unfocussed issue on performance](https://github.com/gatsbyjs/gatsby/issues/24332). I imagine anyone trying to pitch using Gatsby is going to be met with a similar response. 'If Gatsby is so optimised, why does its own site perform so poorly?'

Here is the same set of metrics from Lighthouse (in Incognito tab):

<img width="789" alt="Screenshot 2021-01-20 at 09 19 05" src="https://user-images.githubusercontent.com/161489/105153698-a3226500-5b00-11eb-9cab-a5d171fd7fd6.png">

I know there is great work going on to improve Gatsby Image in relation to these metrics, but surely there are other factors at play here, for example the TBT.

Looking at the blocking in the profiler this happens during Gatsby's booting and React's hydration:

<img width="631" alt="Screenshot 2021-01-20 at 09 31 38" src="https://user-images.githubusercontent.com/161489/105155448-838c3c00-5b02-11eb-847d-826cb21ca988.png">

**Comment by JustFly1984** [comment]
Looks like the case for “barefoot shoemaker”
There is also issues with race conditions, and absence of network failure recovery with crash on first page visit tracked by me for around a year, which is covering an umbrella of seemingly unrelated issues, cos error handling is awful.

**Comment by herecydev** [comment]
There's not really a good excuse unfortunately, but what I can say is gatsby has a number of "starters" you can use to get you going with a good idea of performance. For example, https://www.gatsbyjs.com/starters/gatsbyjs/gatsby-starter-blog, scores near perfectly with lighthouse.

Ultimately, it's what _you and your team_ add that will slow down the scores, at least from my experience.

**Reply by Undistraction** [reply]
I think the problem is that anyone would expect a simple starter to score well. The real question is whether non-trivial real-world applications score well. Certainly scores can be tanked through poorly thought out approaches, but why is Gatsby's own site scoring so poorly? I'd suggest that it's an indicator that there are some fundamental issues with how Gatsby boots, code-splits and relies on full hydration that are fundamentally at odds with Google's metrics. Luckily the metrics used for Core Web Vitals that will begin to effect ranking from May are the easiest to address, but as in the example I gave in the question, it doesn't look good to anyone contemplating using Gatsby.
